svf-llvm/lib/LLVMModule.cpp

```diff
@@ -153,24 +153,25 @@ void LLVMModuleSet::build()
 void LLVMModuleSet::createSVFDataStructure()
 {
     getSVFType(IntegerType::getInt8Ty(getContext()));
-    Set<const Function*> candidateDecls;
-    Set<const Function*> candidateDefs;
+    // Functions need to be retrieved in the order of insertion
+    std::vector<const Function*> candidateDecls;
+    std::vector<const Function*> candidateDefs;
 
     for (Module& mod : modules)
     {
         std::vector<Function*> removedFuncList;
         /// Function
         for (Function& func : mod.functions())
         {
-            if (func.isDeclaration())
+            if (func.isDeclaration() && (FunDefToDeclsMap.find(&func) != FunDefToDeclsMap.end() || mod.getName().str() != Options::ExtAPIInput()))
             {
                 /// if this function is declaration
-                candidateDecls.insert(&func);
+                candidateDecls.push_back(&func);
             }
             else
             {
                 /// if this function is definition
-                if (mod.getName().str() == Options::ExtAPIInput() && FunDefToDeclsMap[&func].empty() && func.getName().str() != "svf__main")
+                if (mod.getName().str() == Options::ExtAPIInput() && FunDefToDeclsMap.find(&func) == FunDefToDeclsMap.end() && func.getName().str() != "svf__main")
```

Comment:
combine this condition and the above one into a method. Use the method as a condtion.